iconservice/deploy/icon_score_deploy_engine.py

```diff
@@ -194,6 +195,8 @@ def _on_deploy(self,
             context.tx = None
 
             self._initialize_score(tx_params.deploy_type, score, params)
+
+            ScoreApiGenerator.check_on_deploy(context, score)
```

Comment:
I think it's better to put this line before the `self._initialize_score()` to prevent the unwanted calling of on_install/on_update.